### ISSUE TITLE
Ability to select extensions providing language features

### ISSUE THREAD

**Issue by matangover** [microsoft/vscode#80889]
Some languages are supported by multiple extensions that provide an overlapping set of language features.

Seems like much effort has been done to support this use case. For some features, the provider's 'score' is used to select which provider to use (e.g. for formatting and syntax highlighting). For other features, results from all providers are merged (e.g. completion, hover, definition, problems, symbols).

My problem is that in some cases I would like to choose which features to use from which extension. For example, I have three extensions that provide overlapping language features for Python: Microsoft's Python, pyright, and mypy extensions. 
 * For hovers, I only like the ones provided by the Python extension (it supports showing documentation while the others don't). Seeing hovers from all three extensions is distracting.
 * For error reporting, I would like to see only the ones from mypy. It is possible to filter the problems view, but it's annoying to filter every time, and filtering just for 'mypy' will ignore errors from other languages I have in my workspace. Without filtering, many duplicate errors are shown.
 * For 'go to definition' and code completion, I would like to use only pyright. By default, results from all three extensions are shown – this requires multiple clicks for every 'go to definition' call because results slightly differ between extensions, so they are shown side by side instead of immediately jumping to the definition.

One solution is to ask every extension author to provide options for disabling some features. This is already done by some extensions but AFAIK not by most. It can be solved centrally by allowing users to selectively disable language features provided by extensions. Something similar has [already been implemented](https://github.com/microsoft/vscode/issues/41882) for formatting, using a dedicated setting. I think it should be extended to other language features. 

Something like:
```json
"languageFeatures.[extension].goToDefinition.enabled": false
```

By default all features from all extensions would be enabled.

**Comment by jrieken** [comment]
Yes, we did this for formatting since it seems a common scenario to have multiple formatter installed (something we didn't anticipate to be honest... why install extension if you don't use them?) You scenario seems more unexpected and needs some explanation. Are these extension soo bad in some areas and soo good in other areas that you need them all? Wouldn't it be better and simpler to have a single extension that fulfils your needs? Is that totally out of reach?

**Comment by matangover** [comment]
Thanks for your response. I agree, in the perfect hypothetical world. But a per-feature on-off switch seems like a fairly straightforward fix that can be applied centrally.

I think we can assume that going forward, language servers will continue to "compete" as languages evolve. It's nice as a user to be able to mix and match their features.

**Comment by matangover** [comment]
The extensions are neither soo bad nor soo good, but I just like my dev environment soo perfect :)

**Comment by hyoretsu** [comment]
> something we didn't anticipate to be honest... why install extension if you don't use them?

Quite old, but I mainly code in JS, but I'm learning C on college and doing some Python things, so I've got quite a lot of extensions for those. Developers aren't limited to a single language.
